services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/logging/DisclosureLogServiceTest.kt

```diff
@@ -91,340 +99,507 @@ internal class DisclosureLogServiceTest {
+                            userRole = UserRole.SERVICE,
+                        )
+                }
+            val capturedLogs = slot<Collection<AuditLogEntry>>()
+            every { auditLogService.createAll(capture(capturedLogs)) } returns mutableListOf()
+
+            disclosureLogService.saveDisclosureLogsForAllu(
+                applicationId,
+                cableReportApplication,
+                expectedStatus
             )
-        verify(exactly = 1) { auditLogService.createAll(any()) }
+
+            assertThat(capturedLogs.captured).hasSameElementsAs(expectedLogs)
+            verify(exactly = 1) { auditLogService.createAll(any()) }
+        }
     }
 
-    @Test
-    fun `saveDisclosureLogsForApplications with company customers and no contacts does nothing`() {
-        disclosureLogService.saveDisclosureLogsForApplications(listOf(), userId)
+    @Nested
+    inner class SaveDisclosureLogsForApplications {
+        @Test
+        fun `saveDisclosureLogsForApplications logs customers and contacts from all applications while ignoring duplicates`() {
+            val contacts =
+                (1..8).map {
+                    ApplicationFactory.createContact(firstName = "Contact", lastName = "$it")
+                }
+            val customers =
+                (1..4).map { ApplicationFactory.createPersonCustomer(name = "Customer $it") }
+            val customersWithContacts =
+                customers.withIndex().map { (i, customer) ->
+                    CustomerWithContacts(
+                        customer,
+                        listOf(contacts[2 * i], contacts[2 * i + 1], contacts[0])
+                    )
+                }
+            val applications =
+                listOf(
+                    ApplicationFactory.createApplication(id = 1, hankeTunnus = hankeTunnus)
+                        .withApplicationData(
+                            customerWithContacts = customersWithContacts[0],
+                            contractorWithContacts = customersWithContacts[1],
+                        ),
+                    ApplicationFactory.createApplication(id = 2, hankeTunnus = hankeTunnus)
+                        .withApplicationData(
+                            customerWithContacts = customersWithContacts[2],
+                            contractorWithContacts = customersWithContacts[3],
+                        )
+                )
+            val capturedLogs = slot<Collection<AuditLogEntry>>()
+            every { auditLogService.createAll(capture(capturedLogs)) } returns mutableListOf()
+
+            disclosureLogService.saveDisclosureLogsForApplications(applications, userId)
+
+            assertThat(capturedLogs.captured)
+                .containsExactlyInAnyOrder(
+                    AuditLogEntryFactory.createReadEntryForCustomer(1, customers[0], HAKIJA),
+                    // The first contact is added twice for the first customer, but identical
+                    // entries
```

Comment:
ktfmt broke this comment to too many lines.

Reply:
Fixed.